lib/elixir/lib/calendar/date.ex

```diff
@@ -943,8 +943,8 @@ defmodule Date do
   @spec year_of_era(Calendar.date()) :: {Calendar.year(), non_neg_integer()}
   def year_of_era(date)
 
-  def year_of_era(%{calendar: calendar, year: year}) do
-    calendar.year_of_era(year)
+  def year_of_era(%{calendar: calendar, year: year, month: month, day: day}) do
+    calendar.year_of_era(year, month, day)
```

Comment:
year_of_era/1 is present on Elixir 1.12 so I think this is a breaking change. Should we go through a deprecation period, in other words something like:

```elixir
if function_exported?(calendar, :year_of_era, 3) do
  calendar.year_of_era(year, month, day)
else
  IO.warn "year_of_era/1 is deprecated in favour of year_of_era/3"
  calendar.year_of_era(year)
end
```